### PR TITLE
feat: hide alert when navigating to protect wallet

### PR DESCRIPTION
**Pull request by tommasini**
**Description**

This PR aims to solve a bug on the backup alert, when navigating to the protect your wallet screen the alert kept being visible.

This PR contains BackupAlert converted to typescript and functional component.

## **Manual testing steps**

_1. Step1: Create a new wallet
_2. Step2: Do not protect your wallet pressing (remember later)
_3. Step3: Press protect your wallet on the backup alert
_4. Step4: The protect your wallet should not be there

## **Screenshots/Recordings**
https://recordit.co/skV2sxdzil

new Bell Icon:
image
image

### **Before**

_[screenshot]_

### **After**

_[screenshot]_

## **Related issues**

_Fixes #4400 

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by plasmacorral** [comment]
Had a chance to take a quick look here and am confident that this addresses the issue reported.  Specifically, tested with the SRP backup demand flow and was pleased with the result.  

Tested commit 55d3aee20d1eab692324a72b31b89aab87ba9956 on Samsung a515f with Android 12.

Manual testing steps for SRP backup demand
1. Create a new wallet
2. Do not protect your wallet pressing (remember later)
3. Import a PK for an account with assets present to trigger backup demand
4. App is unusable for anything but SRP backup
5. Warning is also not presented overtop the SRP backup flow in this condition

File: app/components/UI/BackupAlert/BackupAlert.styles.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { StyleSheet } from 'react-native';
+import { fontStyles } from '../../../styles/common';
+import Device from '../../../util/device';
+import { Theme } from '../../../util/theme/models';
+
+/**
+ * Style sheet function for ModalConfirmation component.
```
Comment (jpcloureiro) [inline app/components/UI/BackupAlert/BackupAlert.styles.ts:7]: update header comment
```suggestion
 * Style sheet function for BackupAlert component.
```

File: app/components/UI/BackupAlert/BackupAlert.styles.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ * @param params Style sheet params.
+ * @param params.theme App theme from ThemeContext.
+ * @param params.vars Inputs that the style sheet depends on.
```
Comment (jpcloureiro) [inline app/components/UI/BackupAlert/BackupAlert.styles.ts:11]: We're not using `params.vars` we can remove this line

File: app/components/UI/BackupAlert/BackupAlert.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+  const shouldNotRenderAlert =
+    seedphraseBackedUp ||
+    inBlockedView ||
+    !backUpSeedphraseVisible ||
+    onboardingWizardStep !== 0 ||
+    !isVisible;
+
+  if (shouldNotRenderAlert) return null;
+  return (
```
Comment (jpcloureiro) [inline app/components/UI/BackupAlert/BackupAlert.tsx:111]:
```suggestion
  return shouldNotRenderAlert ? null
  : (
```

File: app/components/UI/BackupAlert/BackupAlert.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+          : styles.modalViewNotInBrowserView,
+      ]}
+    >
+      <View style={styles.backupAlertWrapper}>
+        <View
+          style={styles.touchableView}
+          {...generateTestId(Platform, SECURE_WALLET_BACKUP_ALERT_MODAL)}
+        >
+          <View style={styles.backupAlertIconWrapper}>
+            {/* We need a icon on our component library to use icon component*/}
```
Comment (jpcloureiro) [inline app/components/UI/BackupAlert/BackupAlert.tsx:127]: Is this comment converted into a ticket to create the icon component?

**Review by jpcloureiro — CHANGES_REQUESTED** [review]

File: app/components/UI/BackupAlert/BackupAlert.tsx (hunk 4 above)
Comment (tommasini) [reply app/components/UI/BackupAlert/BackupAlert.tsx:127]: It was not!!
[Created](https://github.com/MetaMask/metamask-mobile/issues/8406) and sent to the design systems team!
Amazing call!

File: app/components/UI/BackupAlert/BackupAlert.tsx (hunk 3 above)
Comment (tommasini) [reply app/components/UI/BackupAlert/BackupAlert.tsx:111]: Addressed!

File: app/components/UI/BackupAlert/BackupAlert.styles.ts (hunk 2 above)
Comment (tommasini) [reply app/components/UI/BackupAlert/BackupAlert.styles.ts:11]: Removed!

File: app/components/UI/BackupAlert/BackupAlert.styles.ts (hunk 1 above)
Comment (tommasini) [reply app/components/UI/BackupAlert/BackupAlert.styles.ts:7]: Thanks for the review <3

updated!

**Comment by tommasini** [comment]
Actually there is a new Bell Icon already, will ask for design-qa, updated the final result on the description

**Comment by yanrong-chen** [comment]
@tommasini Looks good to me for now! Thank you for checking with me. Later we can also update this to use our DS [banner alert](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?type=design&node-id=5711-16548&mode=design&t=yw3OUBHPHwp8zBOI-0) component

**Review by MarioAslau — APPROVED** [review]
Codewise it looks great Tomas! I dm'd you a couple of questions regarding the context of some choices

**Review by jpcloureiro — APPROVED** [review]
LGTM

**Pull request merged**